src/pybind/ceph_daemon.py

```diff
@@ -0,0 +1,275 @@
+    except Exception as e:
+        raise RuntimeError('exception getting command descriptions: ' + str(e))
+
+    if cmd == 'get_command_descriptions':
+        return cmd_json
+
+    sigdict = parse_json_funcsigs(cmd_json, 'cli')
+    valid_dict = validate_command(sigdict, cmd)
+    if not valid_dict:
+        raise RuntimeError('invalid command')
+
+    if format:
+        valid_dict['format'] = format
+
+    try:
+        ret = do_sockio(asok_path, json.dumps(valid_dict))
+    except Exception as e:
+        raise RuntimeError('exception: ' + str(e))
+
+    return ret
+
+
+class DaemonWatcher(object):
+    """
+    Given a Ceph daemon's admin socket path, poll its performance counters
+    and output a series of output lines showing the momentary values of
+    counters of interest (those with the 'nick' property in Ceph's schema)
+    """
+    (
+        BLACK,
+        RED,
+        GREEN,
+        YELLOW,
+        BLUE,
+        MAGENTA,
+        CYAN,
+        GRAY
+    ) = range(8)
+
+    RESET_SEQ = "\033[0m"
+    COLOR_SEQ = "\033[1;%dm"
+    COLOR_DARK_SEQ = "\033[0;%dm"
+    BOLD_SEQ = "\033[1m"
+    UNDERLINE_SEQ = "\033[4m"
+
+    def __init__(self, asok):
+        self.asok_path = asok
+        self._colored = False
+
+        self._stats = None
+        self._schema = None
+
+    def supports_color(self, ostr):
+        """
+        Returns True if the running system's terminal supports color, and False
+        otherwise.
+        """
+        unsupported_platform = (sys.platform in ('win32', 'Pocket PC'))
+        # isatty is not always implemented, #6223.
+        is_a_tty = hasattr(ostr, 'isatty') and ostr.isatty()
+        if unsupported_platform or not is_a_tty:
+            return False
+        return True
+
+    def colorize(self, msg, color, dark=False):
+        """
+        Decorate `msg` with escape sequences to give the requested color
+        """
+        return (self.COLOR_DARK_SEQ if dark else self.COLOR_SEQ) % (30 + color) \
+               + msg + self.RESET_SEQ
+
+    def bold(self, msg):
+        """
+        Decorate `msg` with escape sequences to make it appear bold
+        """
+        return self.BOLD_SEQ + msg + self.RESET_SEQ
+
+    def format_dimless(self, n, width):
+        """
+        Format a number without units, so as to fit into `width` characters, substituting
+        an appropriate unit suffix.
+        """
+        units = [' ', 'k', 'M', 'G', 'T', 'P']
+        unit = 0
+        while len("%s" % (int(n) / (1000**unit))) > width - 1:
+            unit += 1
+
+        if unit > 0:
+            truncated_float = ("%f" % (n / (1000.0 ** unit)))[0:width - 1]
+            if truncated_float[-1] == '.':
+                truncated_float = " " + truncated_float[0:-1]
+        else:
+            truncated_float = "%{wid}d".format(wid=width-1) % n
+        formatted = "%s%s" % (truncated_float, units[unit])
+
+        assert len(formatted) == width
```

Comment:
why is an assert needed here? Does this really need to break with an `AssertionError` if the `len` doesn't match the `width`? or maybe it was left as debugging?